### PR TITLE
test(view): cover text overflow edge cases

### PR DESCRIPTION
**Pull request by danielraffel**
## Coverage batch

- Adds 38 meaningful assertions in `test/test_text_overflow.cpp`.
- Covers UTF-8 continuation, three-byte, four-byte, and truncated multibyte paths.
- Covers `truncate_to_width` exact-fit, collapsed-box, ellipsis-boundary, and multibyte-prefix behavior using `RecordingCanvas` measurement.
- Addresses Codex P2 by aligning malformed UTF-8 behavior with the documented helper contract: stray continuation bytes are skipped instead of consuming codepoint budget.
- Includes Shipyard's required SDK version bump for the public-header behavior fix.
- Based on latest `origin/main` at `d0f07c495b6612dc66fe95ee1f0729e11e08dcc1`.
- Head SHA: `b6cbc8b655b7558e7fc9fe3d7457faa65f359dbd`.

## Local/lightweight validation

- `git diff --check origin/main...HEAD` passed.
- Assertion batch count confirmed: 38 added `REQUIRE` assertions.
- `tools/scripts/skill_sync_check.py --base origin/main --mode=report` passed.
- `tools/scripts/version_bump_check.py --base origin/main --mode=report` passed after Shipyard applied the SDK bump.
- CLI skew helper ran with no blocking output.

## Shipyard / CI note

Opened through `shipyard pr --no-apply-bumps --skip-target windows --skip-target ubuntu`, then re-run through Shipyard without `--no-apply-bumps` after the P2 fix touched a public header and required an SDK bump. The local mac target failed during configure before build/test because this local runner lacks populated Skia/Dawn for `examples/design-tool` (`pulp-design-tool requires PULP_HAS_SKIA=TRUE`). GitHub Actions Build and Coverage are running for the pushed branch and should be used for branch validation and Codecov freshness.

### CODE REVIEW

**Comment by danielraffel** [comment]
Thanks for the feedback. You're right — the test assertion for the continuation byte should align with the documented behavior of skipping stray continuation bytes. I'll update the expectation to match the utf8_codepoint_count contract.

File: test/test_text_overflow.cpp [hunk 1]
```diff
@@ -93,3 +93,93 @@ TEST_CASE("utf8 helpers count and advance over leading bytes correctly",
     REQUIRE(utf8_advance("\xc3\xb1o", 1) == 2);  // skip past 2-byte ñ
     REQUIRE(utf8_advance("hello", 99) == 5);     // clamps to size
 }
+
+TEST_CASE("utf8 helpers handle continuation three-byte and four-byte paths",
+          "[view][text-overflow][coverage][phase3]") {
+    const std::string stray_continuation("\x80" "abc", 4);
+    REQUIRE(utf8_codepoint_count(stray_continuation) == 4);
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply test/test_text_overflow.cpp]: Thanks for the feedback. You're right — the test assertion for the continuation byte should align with the documented behavior of skipping stray continuation bytes. I'll update the expectation.
